### PR TITLE
Remove 1.25 supported K8s version from e2e cluster

### PR DESCRIPTION
**Pull request by Kalaiselvi84**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, press enter to put that in a new line, and remove leading whitespace from that line:
>

/kind breaking

> /kind bug
> /kind cleanup
> /kind documentation
> /kind feature
> /kind hotfix
> /kind release

**What this PR does / Why we need it**:

**Which issue(s) this PR fixes**:

Work on #3371

**Special notes for your reviewer**:

### CODE REVIEW

**Review by gongmax — APPROVED** [review]

File: test/e2e/framework/framework.go [hunk 1]
```diff
@@ -292,8 +292,7 @@ func (f *Framework) WaitForGameServerState(t *testing.T, gs *agonesv1.GameServer
 // Each Allocated GameServer gets deleted allocDuration after it was Allocated.
 // GameServers will continue to be Allocated until a message is passed to the done channel.
 func (f *Framework) CycleAllocations(ctx context.Context, t *testing.T, flt *agonesv1.Fleet, period time.Duration, allocDuration time.Duration) {
-	// nolint:staticcheck
-	err := wait.PollImmediateUntil(period, func() (bool, error) {
+	err := wait.PollUntilContextCancel(ctx, period, true, func(ctx context.Context) (bool, error) {
```
Comment (gongmax) [inline test/e2e/framework/framework.go:295]: Looks like this doesn't work because PollUntilContextCancel doesn't return ErrWaitTimeout error so error was thrown in line 315. Please revert the change for this line.

Comment (Kalaiselvi84) [reply test/e2e/framework/framework.go:295]: It will be annotated with // nolint:staticcheck?

Comment (gongmax) [reply test/e2e/framework/framework.go:295]: Yes just revert the change you made for this line in this PR

Comment (Kalaiselvi84) [reply test/e2e/framework/framework.go]: Below code will work I guess,
```
func (f *Framework) CycleAllocations(ctx context.Context, t *testing.T, flt *agonesv1.Fleet, period time.Duration, allocDuration time.Duration) {
	err := wait.PollUntilContextCancel(ctx, period, true, func(ctx context.Context) (bool, error) {
		gsa := GetAllocation(flt)
		gsa, err := f.AgonesClient.AllocationV1().GameServerAllocations(flt.Namespace).Create(ctx, gsa, metav1.CreateOptions{})
		if err != nil || gsa.Status.State != allocationv1.GameServerAllocationAllocated {
			// Ignore error. Could be that the buffer was empty, will try again next cycle.
			return false, nil
		}

		// Deallocate after allocDuration.
		go func(gsa *allocationv1.GameServerAllocation) {
			time.Sleep(allocDuration)
			err := f.AgonesClient.AgonesV1().GameServers(gsa.Namespace).Delete(ctx, gsa.Status.GameServerName, metav1.DeleteOptions{})
			require.NoError(t, err)
		}(gsa)

		return false, nil
	})
	if err != nil {
		require.NoError(t, err)
	}
}
```

Comment (Kalaiselvi84) [reply test/e2e/framework/framework.go]: will work on this in the upcoming PR
